### PR TITLE
Prow dashboard: use job name instead of ip address for filtering bosk…

### PR DESCRIPTION
**Pull request by chaodaiG**
…os instances

There are prow instance(s) running boskos, the metrics it emits are from the pod running boskos, the instance name (ip address) changes when the boskos pod is recreated, which makes it impossible to use it as diffentiator. Use job name as differentiator is more reliable and less easy breaking when external boskos ip changes.

Note: this is a breaking change, boskos dashboard will only display values from k8s-infra-prow-builds-boskos after this PR

### CODE REVIEW

**Comment by chaodaiG** [comment]
/assign @cjwagner @spiffxp 
cc @amwat

**Review by cjwagner — APPROVED** [review]
> Note: this is a breaking change, boskos dashboard will only display values from k8s-infra-prow-builds-boskos after this PR

I don't think this is quite what we will see. I would expect 
- the dashboards for the k8s-infra resources to lose their history, but begin displaying data correctly going forward
- the dashboards for the default build cluster resources to display data correctly going forward, but historical data would incorrectly include results from both build clusters (because they previously were recorded with the same job name).

I think this is probably fine since we don't really need to examine the boskos resource graphs for anything besides debugging and quota capacity analysis, both of which are infrequent and neither of which should require more than a couple weeks data.
Hold for confirmation and approval of that assumption.
/hold

**Comment by chaodaiG** [comment]
> * the dashboards for the k8s-infra resources to lose their history, but begin displaying data correctly going forward
> * the dashboards for the default build cluster resources to display data correctly going forward, but historical data would incorrectly include results from both build clusters (because they previously were recorded with the same job name).

You are right on both. I didn't realize that these two boskos instances have the same projects types, which will for sure make the default build cluster resources to be displayed incorrectly

File: config/prow/cluster/monitoring/mixins/lib/config.libsonnet [hunk 1]
```diff
@@ -63,17 +63,17 @@ local config = {
...
-    {instance: "35.225.208.117:9090", type: "gce-project", friendly: "GCE project (k8s-infra)"},
-    {instance: "104.197.27.114:9090", type: "gke-project", friendly: "GKE project"},
-    {instance: "104.197.27.114:9090", type: "gpu-project", friendly: "GPU project"},
-    {instance: "35.225.208.117:9090", type: "gpu-project", friendly: "GPU project (k8s-infra)"},
-    {instance: "104.197.27.114:9090", type: "ingress-project", friendly: "Ingress project"},
-    {instance: "104.197.27.114:9090", type: "node-e2e-project", friendly: "Node e2e project"},
-    {instance: "104.197.27.114:9090", type: "scalability-project", friendly: "Scalability project"},
-    {instance: "35.225.208.117:9090", type: "scalability-project", friendly: "Scalability project (k8s-infra)"},
-    {instance: "104.197.27.114:9090", type: "scalability-presubmit-project", friendly: "Scalability presubmit project"}
+    {job: "k8s-prow-builds-boskos", type: "aws-account", friendly: "AWS account"},
```
Comment (spiffxp) [inline config/prow/cluster/monitoring/mixins/lib/config.libsonnet:66]: if you add `instance` back, the query I suggested above will preserve history.  Could add a TODO to drop in 6 months from current date, when all old data has aged out
```suggestion
    {instance: "104.197.27.114:9090", job: "k8s-prow-builds-boskos", type: "aws-account", friendly: "AWS account"},
```

**Review by spiffxp — COMMENTED** [review]
_Thank you_ for doing this with names, using bare IP's felt really bad.

But, I would like the dashboard to preserve history if it's not too much hassle.

If it is too much hassle, we'll live with the history loss for now (but definitely should send an e-mail notifying sig-release, sig-testing, kubernetes-dev if that's the case)

File: config/prow/cluster/monitoring/mixins/grafana_dashboards/boskos.jsonnet [hunk 2]
```diff
@@ -25,7 +25,7 @@ dashboard.new(
         aliasColors={'busy': '#ff0000', 'cleaning': '#00eeff', 'dirty': '#ff8000', 'free': '#00ff00', 'leased': '#ee00ff', 'other': '#aaaaff', 'toBeDeleted': '#fafa00', 'tombstone': '#cccccc'}
     )
     .addTarget(prometheus.target(
-        std.format('sum(boskos_resources{type="%s",instance="%s"}) by (state)', [resource.type, resource.instance]),
+        std.format('sum(boskos_resources{type="%s",job="%s"}) by (state)', [resource.type, resource.job]),
```
[the remark this answers is not in the document]
Comment (chaodaiG) [reply config/prow/cluster/monitoring/mixins/grafana_dashboards/boskos.jsonnet]: Excellent idea! Done, `instance=""` will work for other instances.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
thanks!

**Pull request merged**